### PR TITLE
Add openshift support

### PR DESCRIPTION
**Pull request by balupton**
As Heroku now requires popular free dynos to be down for 6 hours a day, moving from Heroku to OpenShift is probably something people want to do. This pull request adds support for OpenShift.

Uses https://github.com/bevry/hostenv to fetch the correct PORT and HOSTNAME variables for many different servers (including openshift). OpenShift requires explicit hostname on `.listen`.

Once merged, `rhc app deploy https://github.com/balupton/slackin.git#feature-openshift` should be changed to `rhc app deploy https://github.com/raunchg/slackin.git#master`

To get started with OpenShift today (before this pull request is merged), you can browse the instructions here: https://github.com/balupton/slackin/tree/feature-openshift#openshift

### CODE REVIEW

**Comment by rauchg** [comment]
This looks too complicated. Maybe we can add it as a link to a wiki page? Thanks!

**Comment by balupton** [comment]
@rauchg moved the install instructions to the wiki page, leaving the pull request just for the code changes - if you want me to open a new pull request with just the code changes, let me know

File: package.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
     "debug": "2.1.1",
     "email-regex": "1.0.0",
     "express": "4.11.0",
+    "hostenv": "^1.0.1",
```
Comment (rauchg) [inline package.json]: please pin

**Comment by balupton** [comment]
@rauchg pinned

**Comment by rauchg** [comment]
Thanks! Amazing work

**Pull request merged**